### ISSUE TITLE
XXX has already been defined in YYY

### ISSUE THREAD

**Issue by boldt** [dotnet/docfx#4893]
**Operation System**: Windows 10
**DocFX Version Used**:  2.43.2
**Template used**: default
**Steps to Reproduce**:

* I changed the Namepace of a class from `de.domainname.mysql` (Java style) to `Domainname.Mysql` (C# style).
* Run Build -> Build solution (which builds the doc)

**Expected Behavior**:

No warnings.

**Actual Behavior**:

After I changed the Namespace of a class, I get plenty of warnings like:

```
1>[19-07-18 09:58:34.524]Warning:[BuildCommand.BuildCore.Build Document.LinkPhaseHandlerWithIncremental.ManagedReferenceDocumentProcessor.Save](api/Domainname.Mysql.yml)Uid(Domainname.Mysql) has already been defined in api/de.domainname.mysql.yml.
```

Something seems to be cached, but I have no clue what.

**Comment by superyyrrzz** [comment]
Sorry for the late reply...

docfx has 2 phases:
1. `metadata`: generate YAML files from source code,
2. `build`: build YAML files into HTML output.

From your description, it looks like the old YAMLs and new YAMLs entering into phase2 together. Usually, this should not happen, as phase1 will clear all the old YAMLs in the output folder before generating new YAMLs.

Can these information give you some hints to resolve this?

**Comment by JeremyCaney** [comment]
FWIW: I ran into this issue when using the `DocFx.Console` NuGet package. When rebuilding the project, the `/Api/{Namespace}.{Type}.yml` files remained in their original location. New files were added, but the original files were not deleted. Manually deleting these `*.yml` files after the change resolved the warnings, and they did not appear again.

In my case, I hadn't actually changed the namespace name, but simply subdivided my output folders to e.g., `/Api/{Assembly}/`, which caused a conflict with the original `*.yml` files stored in the `/Api/` directory. 

It's unclear why these aren't deleted as part of the build process, as per @superyyrrzz's comment above. This is obviously a rare scenario, however, and easy to pick up after doing some refactoring. Special handling may be required if your build server maintains temporary files between builds.
